src/code/irqmgr.c

```diff
@@ -1,122 +1,174 @@
-                iter->queue, iter->queue->validCount);
+                client->queue, MQ_GET_COUNT(client->queue));
         } else {
-            osSendMesg(iter->queue, msg, OS_MESG_NOBLOCK);
+            osSendMesg(client->queue, msg, OS_MESG_NOBLOCK);
         }
-
-        iter = iter->prev;
     }
 }
 
-void IrqMgr_JamMesgForClient(IrqMgr* this, OSMesg msg) {
-    IrqMgrClient* iter = this->clients;
-
-    while (iter != NULL) {
-        if (iter->queue->validCount >= iter->queue->msgCount) {
+/**
+ * Send `msg` to every registered client if the message queue is not full. This appears to be for
+ * high-priority messages that should be jammed to the front of the queue, however a bug prevents
+ * this from working in this way and the message is appended to the back of the queue as in
+ * `IrqMgr_SendMesgToClients`.
+ *
+ * @see IrqMgr_SendMesgToClients
+ */
+void IrqMgr_JamMesgToClients(IrqMgr* this, OSMesg msg) {
+    IrqMgrClient* client;
+
+    for (client = this->clients; client != NULL; client = client->prev) {
+        if (MQ_IS_FULL(client->queue)) {
             // "irqmgr_JamMesgForClient: Message queue is overflowing mq=%08x cnt=%d"
             osSyncPrintf(
                 VT_COL(RED, WHITE) "irqmgr_JamMesgForClient:メッセージキューがあふれています mq=%08x cnt=%d\n" VT_RST,
-                iter->queue, iter->queue->validCount);
+                client->queue, MQ_GET_COUNT(client->queue));
         } else {
-            // mistake? the function's name suggests it would use osJamMesg
-            osSendMesg(iter->queue, msg, OS_MESG_NOBLOCK);
+            //! @bug The function's name suggests this would use osJamMesg rather than osSendMesg, using the
+            //! latter makes this function no different than IrqMgr_SendMesgToClients.
+            osSendMesg(client->queue, msg, OS_MESG_NOBLOCK);
         }
-        iter = iter->prev;
     }
 }
 
+/**
+ * Runs when the Pre-NMI OS Event is received. This indicates that the console will reset in 0.5s / 500ms.
+ * Updates the reset status and time before forwarding the Pre-NMI message to registered clients so they
+ * may begin shutting down in advance of the reset.
+ */
 void IrqMgr_HandlePreNMI(IrqMgr* this) {
-    u64 temp = STATUS_PRENMI; // required to match
+    u64 preNmi = IRQ_RESET_PRENMI; // required to match
 
-    gIrqMgrResetStatus = temp;
-    this->resetStatus = STATUS_PRENMI;
+    gIrqMgrResetStatus = preNmi;
+    this->resetStatus = IRQ_RESET_PRENMI;
     sIrqMgrResetTime = this->resetTime = osGetTime();
 
-    osSetTimer(&this->timer, OS_USEC_TO_CYCLES(450000), 0ull, &this->queue, (OSMesg)PRENMI450_MSG);
-    IrqMgr_JamMesgForClient(this, (OSMesg) & this->prenmiMsg);
+    // Schedule a PRENMI450 message to be handled in 450ms
+    osSetTimer(&this->timer, OS_MSEC_TO_CYCLES(450), 0, &this->queue, (OSMesg)PRENMI450_MSG);
+    IrqMgr_JamMesgToClients(this, (OSMesg) & this->prenmiMsg);
```

Comment:
```suggestion
    IrqMgr_JamMesgToClients(this, (OSMesg)&this->prenmiMsg);
```
This is surprisingly confusing-looking.

Reply:
Yeah.. the formatter has some weird ideas about these

Reply:
I can sort of see its point, but still...